vendor/src/github.com/imdario/mergo/merge.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2013 Dario Castañé. All rights reserved.
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Based on src/pkg/reflect/deepequal.go from official
+// golang's stdlib.
+
+package mergo
+
+import (
+	"reflect"
+)
+
+// Traverses recursively both values, assigning src's fields values to dst.
+// The map argument tracks comparisons that have already been seen, which allows
+// short circuiting on recursive types.
+func deepMerge(dst, src reflect.Value, visited map[uintptr]*visit, depth int, overwrite bool) (err error) {
+	if !src.IsValid() {
+		return
+	}
+	if dst.CanAddr() {
+		addr := dst.UnsafeAddr()
+		h := 17 * addr
```

Comment:
Awesome, @imdario, basically, we prefer to vendor a tag instead of a specific commit, so I was wondering if you're planning on tagging a new release, because your master branch has some bug fixes. See the comment above (hope it works, because it's a comment on an outdated diff; https://github.com/docker/docker/pull/18587#discussion_r49614276)